extensions/azure-extensions/src/main/java/io/druid/storage/azure/AzureLoadSpec.java

```diff
@@ -31,24 +31,24 @@
 public class AzureLoadSpec implements LoadSpec
 {
   @JsonProperty
-  private final String storageDir;
+  private final String blobPath;
 
   private final AzureDataSegmentPuller puller;
 
   @JsonCreator
   public AzureLoadSpec(
-      @JacksonInject AzureDataSegmentPuller puller,
-      @JsonProperty("storageDir") String storageDir
```

Comment:
Some descriptions from [How to use Blob storage from Java](http://azure.microsoft.com/en-us/documentation/articles/storage-java-how-to-use-blob-storage/):

`Storage Account`: All access to Azure Storage is done through a storage account. See Azure Storage Scalability and Performance Targets for details about storage account capacity.

`Container`: A container provides a grouping of a set of blobs. All blobs must be in a container. An account can contain an unlimited number of containers. A container can store an unlimited number of blobs.

`Blob`: A file of any type and size. There are two types of blobs that can be stored in Azure Storage: block and page blobs. Most files are block blobs. A single block blob can be up to 200 GB in size. This tutorial uses block blobs. Page blobs, another blob type, can be up to 1 TB in size, and are more efficient when ranges of bytes in a file are modified frequently. For more information about blobs, see Understanding Block Blobs and Page Blobs.

Given that, `account` and `key` are tied to authentication as @xvrl mentioned before, `container` is just a group of blobs (there is also a definition of `virtual directory` here).

So, +1 to add `container` to `AzureLoadSpec` (and not add `account`), due to the following reason:

> I think having the container be part of the loadSpec would help future-proof azure deep-storage module and avoid having to introduce ugly backwards-compatibility fixes when we want to support multiple containers or moving data between containers.